### ISSUE TITLE
links {} does not seem to handle external static libraries properly

### ISSUE THREAD

**Issue by AndrewJDR** [bkaradzic/GENie#96]
First some references:
http://stackoverflow.com/questions/10482997/linking-to-an-external-library-with-premake
One of the premake devs responds that it is a bug and mentions this commit in premake which supposedly fixes the issue:
https://github.com/annulen/premake/commit/c144e9c4f30505495a562d9cfdb8ff9ef15c2495
I'm not sure how relevant those are to genie, but thought I should mention it right off the bat.

Repro with genie on linux:
- Clone bx/bgfx
- Open bgfx/scripts/genie.lua and find the first links section within the exampleProject function. It should look like this:

```
­­­­­­links {
"bgfx",
"example-common",
­­­­­­­}
```
- Make it look like this:

```
­­­­­­links {
­­­­­­­­­­­­"/tmp/somelib.a",
"bgfx",
­­­­­­"example-common",
­­­­­­­}
```
- "touch /tmp/somelib.a", or put an actual static library file there if you want.
- cd bgfx && make
- Examine .build/projects/gmake-linux/example-00-helloworld.make and search for "somelib.a"

Actual Result in example-00-helloworld.make:
LIBS      += $(LDDEPS) -lrt -lsomelib.a -lX11 -lGLESv2 -lEGL -lbcm_host -lvcos -lvchiq_arm -lpthread

Expected result would be for genie to understand that I'm referring to an external static lib and have something like:
LIBS      += $(LDDEPS) -lrt /tmp/somelib.a -lX11 -lGLESv2 -lEGL -lbcm_host -lvcos -lvchiq_arm -lpthread
Or this:
LIBS      += $(LDDEPS) -lrt -L/tmp somelib.a -lX11 -lGLESv2 -lEGL -lbcm_host -lvcos -lvchiq_arm -lpthread

It's possible to add /tmp/somelib.a to a linkoptions section instead of using links, but linkoptions are always before all of the "links" in the command line, so you then lose control over the ordering of your libraries, which is sometimes important. For example, you may want /tmp/somelib.a to appear after all of the previously specified "links", and as far as I know that is not possible by using linkoptions. The workaround I used in my project was to switch everything over to using linkoptions -- even for in-project static and dynamic libraries that have no other reason to be specified within linkoptions.

**Comment by ajohnson23** [comment]
Ok, there's actually another way of handling this, assuming instead of /tmp/somelib.a you have /tmp/libsomelib.a:
```
libdir {
/tmp
}
­­­­­­links {
­­­­­­­­­­­­"somelib",
"bgfx",
­­­­­­"example-common",
­­­­­­­}
```
This will produce "-L/tmp -lsomelib", which works fine even with static libraries. I suppose the library needs to follow the lib*.a convention, which is not a huge deal for my project. There my be other downsides to this approach of using that I haven't thought of, though.

In any case, this isn't a concern for me anymore, so feel free to close, leave open, or do whatever you want with this issue :)

**Comment by bkaradzic** [comment]
I'll take a look at that patch.

**Comment by ajohnson23** [comment]
There's a downside to my workaround: The resulting makefile does end up not referencing the .a file through any of variables of the the following line:
```
$(TARGET): $(GCH) $(OBJECTS) $(LDDEPS) $(RESOURCES) | $(TARGETDIR) $(OBJDIRS)
```
This means that if the .a file gets updated, the project won't get relinked, which is not ideal.

I *think* that the proper solution for all this is to have logic within genie that checks for .a files within links {} blocks and doesn't prepend the -l in that case, as well as adding the .a file to the $(TARGET) line. I'm opening a PR shortly that implements this.

**Comment by lionleaf** [comment]
This seems to be have been solved by #125 and can be closed?
